### ISSUE TITLE
Support more parameter types in COM interface declarations

### ISSUE THREAD

**Issue by Herschel** [microsoft/windows-rs#1554]
### Motivation

COM interface declaration support was initially added in #1540. Currently it only supports [parameters of pointer type](https://github.com/microsoft/windows-rs/blob/e271fb89573ba1bd712fbe8862ebb6a4b2c9b891/crates/libs/interface/src/lib.rs#L105-L107) which can be passed through the FFI boundary.

For example, the following fails:
```
#[interface("00000000-0000-0000-0000-000000000000")]
unsafe trait ITest: IUnknown {
    fn SetFoo(&self, i: i32) -> HRESULT;
}
```

This should support more parameter types, including primitive types and common Windows API types which can also be passed through based on something like `AbiTransferable` in com-rs.

(Naturally this is all a very recent addition, still under development and not expected to be working yet -- submitting this for tracking purposes. Thank you!)

**Comment by kennykerr** [comment]
Yep, that was only the start. I hope to support the same type-system as the rest of the `windows` crate supports.

**Comment by kennykerr** [comment]
A few initial issues with the `interface` macro that I'm working through:

* The `interface` macro only lets you write methods with a single parameter. That's obviously an issue. 😉 
* The `interface` will fail if the `implement` feature is not enabled - it makes invalid cross-feature dependencies. It should not need the `implement` feature at first glance, but it may be simpler to combine them...

**Comment by Herschel** [comment]
#1486 was re-opened, so closing this in favor of that issue. Thank you for the efforts!

**Comment by kennykerr** [comment]
Improvements to the `interface` macro are now published as part of [0.34.0](https://github.com/microsoft/windows-rs/releases/tag/0.34.0).
